### PR TITLE
Add perma-links/references to hash-based content directories

### PR DESCRIPTION
**Pull request by nicklauslittle**
This PR adds the `$(GetAirpowerPath)\ref` directory. This directory consists of directory junctions from the tag to the hash-based content directories. It allows users to add something like `$(GetAirpowerPath)\ref\git\cmd` to their `$Path`, so that it always points to the latest version of `git` installed by Airpower.

### CODE REVIEW

**Comment by troyaf** [comment]
Is this a replacement of `install` concept which makes the persistent batch file?

File: src/package.ps1 [hunk 1]
```diff
@@ -102,6 +102,14 @@ function AsPackage {
 	throw "failed to parse package: $Pkg"
 }
 
+function ResolvePackageRefPath {
+	param (
+		[Parameter(Mandatory, ValueFromPipeline)]
+		[object]$Pkg
```
Comment (troyaf) [inline src/package.ps1:108]:
```suggestion
		[Collections.Hashtable]$Pkg
```

Will you update `ResolveRemoteRef`'s param as well?

**Comment by nicklauslittle** [comment]
> Is this a replacement of the `install` concept which makes the persistent batch file?

Yes. It also will probably make things like the `-ScriptBlock` pull request obsolete.

File: src/package.ps1 [hunk 2]
```diff
@@ -303,6 +315,7 @@ function RemovePackage {
 	if ($null -ne $err) {
 		throw $err
 	}
+	Remove-Item ($Pkg | ResolvePackageRefPath) -ErrorAction Ignore
```
Comment (nicklauslittle) [inline src/package.ps1:318]: I wasn't sure if here or at the end of the `UninstallPackage` was a better spot for this. Thoughts?

File: src/package.ps1 (hunk 1 above)
Comment (nicklauslittle) [reply src/package.ps1]: Done. Yeah, I wasn't sure which one it was supposed to be, since some were `[object]` and some were `[Collections.Hashtable]`.

**Review by troyaf — APPROVED** [review]

**Pull request merged**